### PR TITLE
First commit

### PR DESCRIPTION
**Pull request by khellan**
Fully working basic search example.

### CODE REVIEW

File: simple-search/service.py [hunk 1]
```diff
@@ -0,0 +1,44 @@
+from authority import API_KEY
+from json import dumps, loads
+from logging import error
+from requests import get
+
+
+class Endpoint:
```
Comment (oyevtushok) [inline simple-search/service.py]: `class SearchService`?

File: simple-search/service.py [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+        'name': result['name'],
+        'snippet': result['query_snippet'] if 'query_snippet' in result else ''
+    }
+
+
+def format_response(raw_result):
+    parsed_result = loads(raw_result)
+    result = parsed_result['result']
+    docs = result['docs']
+    resultset = [format_result(doc) for doc in docs]
```
Comment (oyevtushok) [inline simple-search/service.py:29]: `formatted_results`?

File: simple-search/static/index.html [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+        <title>minigrow</title>
+        <link href="https://fonts.googleapis.com/css?family=Comfortaa" rel="stylesheet" />
+        <link rel="stylesheet" type="text/css" href="styles.css" />
+    </head>
+    <body>
+        <script src="//unpkg.com/mithril/mithril.js"></script>
+        <script>
+        var query = ''
+        var resultset = []
+        var ResultSet = {
```
Comment (oyevtushok) [inline simple-search/static/index.html:12]: should be different name, previous variable has the same name, so I am not sure what is the difference between those two

File: simple-search/service.py [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+    return resultset
+
+
+def search(query, country=''):
+    parameters = {}
+    if query:
+        parameters['q'] = query
+    if country:
+        parameters['country'] = country
```
Comment (oyevtushok) [inline simple-search/service.py:38]: newline after?

File: simple-search/service.py [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+    response = get(Endpoint.SEARCH, headers=HEADERS, params=parameters)
+    status_code = response.status_code
```
Comment (oyevtushok) [inline simple-search/service.py]: newline after?

File: simple-search/service.py [hunk 6]
```diff
@@ -0,0 +1,44 @@
...
+    if status_code != 200:
+        error('Request to {} status code {}: {}'.format(Endpoint.SEARCH, status_code, response.text))
+        return "Error handling request!"
```
Comment (oyevtushok) [inline simple-search/service.py]: newline after?

**Review by oyevtushok — COMMENTED** [review]
I stared doing the review, but now see that as an example it seems to be a bit overcomplicated... 

* there is no need to use cherrypy - use any webserver you want(it's just a static content, where you can edit one js file for everything)
* there is no need to use `mithril` - pure js or jQuery would be a lot more easier to read, not sure if everyone is excited about it

Then what you will need to add your api key and start your favourite webserver to serve static content.

**Comment by osblinnikov** [comment]
For my eye it is a bit overcomplicated solution to be presented as usage example.

File: simple-search/service.py (hunk 1 above)
Comment (khellan) [reply simple-search/service.py]: Makes sense. Will change.

File: simple-search/service.py (hunk 2 above)
Comment (khellan) [reply simple-search/service.py:29]: Function names should be verbs. Do you think of these more like a property?

File: simple-search/service.py (hunk 4 above)
Comment (khellan) [reply simple-search/service.py:38]: Probably. Will fix.

Comment (khellan) [inline simple-search/service.py]: Ah you mean at the end of the file, why? I don't really have an opinion about it.

File: simple-search/static/index.html (hunk 3 above)
Comment (khellan) [reply simple-search/static/index.html:12]: The point is they are kind of the same. This is a mithril artifact

**Comment by khellan** [comment]
Agree that it might be a bit too complicated. To me it was a point to have something that would resemble an actual application.

No reason to use cherrypy. I chose it for it's simplicity.

I originally started using jquery, but felt that the world had moved beyond it so I chose mithril once again for it's simplicity.

Since the both of you think this was too complicated, should we just scrap it?

Alternatively convert to no js and just static HTML...

File: simple-search/service.py (hunk 4 above)
Comment (khellan) [reply simple-search/service.py:38]: Or actually. No. Why? Default PyCharm formatting would not add a newline. The function is not long. I don't see a newline making sense there.

File: simple-search/service.py (hunk 5 above)
Comment (khellan) [reply simple-search/service.py]: Why?

File: simple-search/service.py (hunk 6 above)
Comment (khellan) [reply simple-search/service.py]: Once again why?

**Comment by vincentp** [comment]
What is this project, is it going to be public examples of our API, part of our doc?

**Comment by khellan** [comment]
It was meant to be a public example. Available in Github and possibly accompanied with a blog post.

File: simple-search/service.py [hunk 7]
```diff
@@ -0,0 +1,44 @@
+from authority import API_KEY
+from json import dumps, loads
+from logging import error
+from requests import get
+
+
+class SearchService:
+    SEARCH = 'https://openapi.companybooknetworking.com/1.0/search/search?'
```
Comment (vincentp) [inline simple-search/service.py:8]: indent? also shouldn't this variable be called ENDPOINT ?

File: simple-search/README.md [hunk 8]
```diff
@@ -0,0 +1,12 @@
...
+
+This is an example of a basic business search using Companybook.
+
+To run it:
+- Create a file called authority.py in this directory.
+- Put API_KEY='<your Companybook API key>' in the authority file.
+- Install cherrypy
+- run using python server.py
+
+This is example works with python 2.7 and python 3.
```
Comment (vincentp) [inline simple-search/README.md]: remove 'is'

File: simple-search/service.py (hunk 7 above)
Comment (khellan) [reply simple-search/service.py:8]: I was thinking about that when I made the change O2 suggested. In principle I agree. Then there are different search service endpoints and this particular one is actually called search. If I used the match endpoint, would it be called ENDPOINT?

More importantly, if we supported both match and search, what would it be called?

**Comment by vincentp** [comment]
I don't see why mixing up two different languages for an example (dev understand the key needs to be secure). Instead, we should build an example just in python or JS.

File: simple-search/service.py (hunk 7 above)
Comment (vincentp) [reply simple-search/service.py:8]: SEARCH_URL and MATCH_URL then

**Comment by khellan** [comment]
Then we scrap this example. I don't want to do a pure JS example since then you end up with the key being public (or a very complicated example)

I will submit a straightforward and simple Python example soon.

**Pull request closed without merging**